Source/WebKit/UIProcess/ProcessThrottler.cpp

```diff
@@ -191,8 +191,10 @@ void ProcessThrottler::setThrottleState(ProcessThrottleState newState)
             weakThis->assertionWasInvalidated();
     });
 
-    if (m_assertion->type() == ProcessAssertionType::Suspended)
+    if (m_assertion->type() == ProcessAssertionType::Suspended && !m_shouldDropSuspendedAssertionImmediately)
```

Comment:
I think I would have factored this like so:
```
if (m_assertion->type() == ProcessAssertionType::Suspended) {
    if (m_shouldDropSuspendedAssertionImmediately)
        m_assertion = nullptr;
    else
        m_removeAllAssertionsTimer.startOneShot(removeAllAssertionsTimeout);
} else
    m_removeAllAssertionsTimer.stop();
```

To avoid checking `m_assertion->type() == ProcessAssertionType::Suspended` twice.